src/boot/efi/console.c

```diff
@@ -114,51 +117,32 @@ EFI_STATUS console_key_read(UINT64 *key, UINT64 timeout_usec) {
         return EFI_SUCCESS;
 }
 
-static EFI_STATUS change_mode(UINTN mode) {
+static EFI_STATUS change_mode(UINT32 mode) {
         EFI_STATUS err;
+        UINT32 old_mode = ST->ConOut->Mode->Mode;
```

Comment:
as mentioned, maybe add an assert() here before doing this implicit conversion from INT32 → UINT32, that checks that ST->ConOut->Mode->Mode is >= 0